engine/dlib/src/dlib/log.cpp

```diff
@@ -189,24 +196,34 @@ static dmSocket::Result SendAll(dmSocket::Socket socket, const char* buffer, int
 
 static void dmLogUpdateNetwork()
 {
-    dmLogServer* self = g_dmLogServer;
-    if (self->m_ServerSocket == dmSocket::INVALID_SOCKET_HANDLE)
+    dmSocket::Socket server_socket = dmSocket::INVALID_SOCKET_HANDLE;
+    bool connections_full = false;
+    {
+        DM_SPINLOCK_SCOPED_LOCK(g_LogServerLock);
+        if (!IsServerInitialized())
+            return;
```

Comment:
Having locks sprinkled throughout the code isn't very nice looking, but we also want to avoid holding the lock for very long.